rest/route/annotations.go

```diff
@@ -247,12 +247,26 @@ func (h *annotationByTaskPutHandler) Factory() gimlet.RouteHandler {
 func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
 	var err error
 	h.taskId = gimlet.GetVars(r)["task_id"]
+	taskExecutionsAsString := gimlet.GetVars(r)["execution"]
```

Comment:
I think you might need `r.URL.Query().Get("execution")` instead of GetVars here.